lib/usecases/base/base_use_case.dart

```diff
@@ -23,3 +23,9 @@ abstract class SimpleUseCase<T> {
 
   T call();
 }
+
+abstract class DataSaverUseCase<T> {
+  const DataSaverUseCase();
+
+  void save(T data);
+}
```

Comment:
I don't think `DataSaverUseCase` should be a base component here. It's actually the `SimpleUseCase` with the call action to save the data.

Reply:
I think he created a new one, because `SimpleUseCase` doesn't provide any input (parameters)

What about we rename this to `SimpleInputUseCase`? 💭

Reply:
That's why I suggested to combine the data-saving action to LogInUseCase 💭